File: 0229-majority-element-ii/0229-majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
    List<Integer> result=new ArrayList<>();
        
        if(nums.length==1)
        {
            result.add(nums[0]);
            return result;
        }
            
        HashMap<Integer,Integer> map=new HashMap<>();
        
        for(int i=0;i<nums.length;i++)
        {
            
            if(map.containsKey(nums[i]))
            {
                map.replace(nums[i],map.get(nums[i])+1);
            }
            else
            {
                map.put(nums[i],1);
            }
        }
        for (Integer i : map.keySet())
        {
            if(map.get(i)>nums.length/3)
                result.add(i);
        }
        
        return result;
    }    
    
}

File: 0229-majority-element-ii/0229-majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        int n=nums.size();
        //if(n<3) return nums;
        unordered_map<int,int> m;
        for(int a:nums)
            m[a]++;
        set<int> res;
        for(int i=0;i<n;++i)
        {
            if(m[nums[i]]>n/3)
                res.insert(nums[i]);
        }
        vector<int> t(res.begin(),res.end());
        return t;
    }
};